python/cudf/cudf/utils/gpu_utils.py

```diff
@@ -7,7 +7,7 @@ def validate_setup(check_dask=True):
     # TODO: Remove the following check once we arrive at a solution for #4827
     # This is a temporary workaround to unblock internal testing
     # related issue: https://github.com/rapidsai/cudf/issues/4827
-    if not check_dask and "DASK_PARENT" in os.environ:
+    if not check_dask and "RAPIDS_NO_INITIALIZE" in os.environ:
```

Comment:
I think we can remove the `check_dask` portion of this as well. The environment variable is now entirely opt-in.